go/cmd/vtctldclient/command/movetables.go

```diff
@@ -0,0 +1,541 @@
+	}
+)
+
+var (
+	// Required options for all commands.
+	moveTablesOptions = struct {
+		Workflow       string
+		TargetKeyspace string
+		Format         string
+	}{}
+	moveTablesCancelOptions = struct {
+		KeepData         bool
+		KeepRoutingRules bool
+	}{}
+	moveTablesCompleteOptions = struct {
+		KeepData         bool
+		KeepRoutingRules bool
+		RenameTables     bool
+		DryRun           bool
+	}{}
+	moveTablesCreateOptions = struct {
+		Workflow                     string
+		SourceKeyspace               string
+		Cells                        []string
+		TabletTypes                  []topodatapb.TabletType
+		TabletTypesInPreferenceOrder bool
+		SourceShards                 []string
+		ExternalClusterName          string
+		AllTables                    bool
+		IncludeTables                []string
+		ExcludeTables                []string
+		SourceTimeZone               string
+		OnDDL                        string
+		DeferSecondaryKeys           bool
+		AutoStart                    bool
+		StopAfterCopy                bool
+	}{}
+	moveTablesSwitchTrafficOptions = struct {
+		Cells                    []string
+		TabletTypes              []topodatapb.TabletType
+		MaxReplicationLagAllowed time.Duration
+		EnableReverseReplication bool
+		Timeout                  time.Duration
+		DryRun                   bool
+		Direction                workflow.TrafficSwitchDirection
+	}{}
+)
+
+func bridgeMoveTablesToWorkflow(cmd *cobra.Command, args []string) {
+	workflowUpdateOptions.Workflow = moveTablesOptions.Workflow
+	workflowOptions.Keyspace = moveTablesOptions.TargetKeyspace
+}
+
+func commandMoveTablesCreate(cmd *cobra.Command, args []string) error {
+	cli.FinishedParsing(cmd)
+
+	tsp := tabletmanagerdatapb.TabletSelectionPreference_ANY
+	if moveTablesCreateOptions.TabletTypesInPreferenceOrder {
+		tsp = tabletmanagerdatapb.TabletSelectionPreference_INORDER
+	}
+
+	req := &vtctldatapb.MoveTablesCreateRequest{
+		Workflow:                  moveTablesOptions.Workflow,
+		TargetKeyspace:            moveTablesOptions.TargetKeyspace,
+		SourceKeyspace:            moveTablesCreateOptions.SourceKeyspace,
+		Cells:                     moveTablesCreateOptions.Cells,
+		TabletTypes:               moveTablesCreateOptions.TabletTypes,
+		TabletSelectionPreference: tsp,
+		AllTables:                 moveTablesCreateOptions.AllTables,
+		IncludeTables:             moveTablesCreateOptions.IncludeTables,
+		ExcludeTables:             moveTablesCreateOptions.ExcludeTables,
+		OnDdl:                     moveTablesCreateOptions.OnDDL,
+		AutoStart:                 moveTablesCreateOptions.AutoStart,
+		StopAfterCopy:             moveTablesCreateOptions.StopAfterCopy,
+	}
+
+	resp, err := client.MoveTablesCreate(commandCtx, req)
+	if err != nil {
+		return err
+	}
+
+	format := strings.ToLower(strings.TrimSpace(moveTablesOptions.Format))
+	switch format {
+	case "text", "json":
+	default:
+		return fmt.Errorf("invalid output format, got %s", moveTablesOptions.Format)
+	}
```

Comment:
if we do this elsewhere and i've forgotten then ignore me, but ...

`cli.FinishedParsing` should be called after any errors that fall under "you specified some input on the command-line that is invalid", and so this should (imo) be moved up to before that call

Reply:
You're right. I'll move this.